python/setup.py

```diff
@@ -138,11 +138,13 @@
     "grpcio >= 1.28.1",
     "jsonschema",
     "msgpack >= 1.0.0, < 2.0.0",
-    "numpy >= 1.16",
+    "numpy >= 1.16; python_version < '3.9'",
+    "numpy >= 1.19.3; python_version >= '3.9'",
     "protobuf >= 3.8.0",
     "py-spy >= 0.2.0",
     "pyyaml",
     "requests",
+    "setproctitle",
```

Comment:
Looks like setproctitle only has linux wheels, so this may still cause `pip install ray` to give compilation errors in a lot of settings.

Reply:
Yeah it seems like we are seeing that issue in the above comments b/t me and @amogkam wrt to the windows build. Surprisingly the macos wheels build perfectly fine, which leads me to believe there is something wrong with our macos wheel test, but not sure. I've decided to hold off until setproctitle publishes their windows wheels (merged in their master just haven't published them yet), since I don't really have the bandwidth/capacity to look more into the source build errors we face with it.

Reply:
update: windows wheels have been added for setproctitle on pypi: https://pypi.org/project/setproctitle/#files but I don't see macos wheels.

Reply:
Now that windows wheels are out, maybe we won't have the same issues with the windows source build I faced earlier. I have created a PR that just specifically updates the setproctitle version. https://github.com/ray-project/ray/pull/14538

Reply:
With #14538 merged in, this is not an issue anymore.